src/button-group/internal.tsx

```diff
@@ -0,0 +1,227 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useImperativeHandle, useRef, forwardRef, useEffect } from 'react';
+import { getBaseProps } from '../internal/base-component';
+import { ButtonGroupProps, InternalButtonGroupProps } from './interfaces';
+import { ButtonProps } from '../button/interfaces';
+import ItemElement from './item-element.js';
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+import clsx from 'clsx';
+import { ButtonDropdownProps } from '../button-dropdown/interfaces';
+import {
+  SingleTabStopNavigationAPI,
+  SingleTabStopNavigationProvider,
+} from '../internal/context/single-tab-stop-navigation-context';
+import { useMergeRefs } from '../internal/hooks/use-merge-refs';
+import { KeyCode } from '../internal/keycode';
+import { hasModifierKeys } from '../internal/events';
+import { circleIndex } from '../internal/utils/circle-index';
+import { getAllFocusables } from '../internal/components/focus-lock/utils';
+import { nodeBelongs } from '../internal/utils/node-belongs';
+import handleKey from '../internal/utils/handle-key';
+
+const InternalButtonGroup = forwardRef(
+  (
+    {
+      items = [],
+      onItemClick,
+      ariaLabel,
+      ariaLabelledby,
+      __internalRootRef = null,
+      ...props
+    }: InternalButtonGroupProps,
+    ref: React.Ref<ButtonGroupProps.Ref>
+  ) => {
+    const baseProps = getBaseProps(props);
+    const focusedItemIdRef = useRef<null | string>(null);
+    const navigationAPI = useRef<SingleTabStopNavigationAPI>(null);
+    const containerObjectRef = useRef<HTMLDivElement>(null);
+    const containerRef = useMergeRefs(containerObjectRef, __internalRootRef);
+    const itemsRef = useRef<Record<string, ButtonProps.Ref | null>>({});
+
+    useImperativeHandle(ref, () => ({
+      focus: id => {
+        itemsRef.current[id]?.focus();
+      },
+    }));
+
+    const onSetButtonRef = (item: ButtonGroupProps.Item, element: ButtonProps.Ref | null) => {
+      const isItemGroup = (item: ButtonDropdownProps.ItemOrGroup): item is ButtonDropdownProps.ItemGroup => {
+        return 'items' in item;
+      };
+
+      const getAllIdsItemOrGroup = (item: ButtonDropdownProps.ItemOrGroup): string[] => {
+        if (isItemGroup(item)) {
+          const values = item.items.flatMap(getAllIdsItemOrGroup);
+          if (typeof item.id !== 'undefined') {
+            values.push(item.id);
+          }
+
+          return values;
+        } else {
+          return [item.id];
+        }
+      };
+
+      const getAllIds = (item: ButtonGroupProps.Item) => {
+        if (item.type === 'icon-button') {
+          return [item.id];
+        } else {
+          const values = getAllIdsItemOrGroup(item);
```

Comment:
Why is it necessary to collect IDs of the menu items?

Reply:
Thes code was removed